### ISSUE TITLE
OpenAPI C# client generator generates wrong request media type for corresponding deserializer when multiple response content possible.

### ISSUE THREAD

**Issue by GimpArm** [dotnet/aspnetcore#40197]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

ASP.NET MVC + Swashbuckle generates a swagger.json for a String response with the content types of `text/plain`, `application/json`, and `text/json` which is perfectly valid, but then the openAPI client generator does not handle this correctly. The client is generated with an accepts header of `text/plain` but then it uses `ReadObjectResponseAsync<string>()` to try to deserialize the plain text response which fails. So the request mime type does not match the deserialization method used.

When the swagger.json only contains `text/plain` then the client is generated with the correct deserializer of `response_.Content.ReadAsStringAsync()`.

If the request accepts header is changed it `application/json` then the existing `ReadObjectResponseAsync<string>()` will work because then the server wraps the string in double quotes (`"`).

_Note: This is when using option /WrapResponses:false_

### Expected Behavior

Use the correct deserializer for the requesting mime type.

### Steps To Reproduce

Basic swagger.json produced from APS.NET MVC for a string or Task&lt;string&gt; response.

```
{
  "openapi": "3.0.1",
  "info": {
    "title": "ExampleService",
    "version": "1.0.0"
  },
  "paths": {
    "/Example/TestString": {
      "get": {
        "tags": [
          "Example"
        ],
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "string"
                }
              },
              "application/json": {
                "schema": {
                  "type": "string"
                }
              },
              "text/json": {
                "schema": {
                  "type": "string"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": { }
}
```

This is the method that is generated by the openAPI client. Notice the `text/plain` added to the header and the using of `ReadObjectResponseAsync<string>()` for deserialization.

```
        public async System.Threading.Tasks.Task<string> TestStringAsync(System.Threading.CancellationToken cancellationToken)
        {
            var urlBuilder_ = new System.Text.StringBuilder();
            urlBuilder_.Append("Example/TestString");
    
            var client_ = _httpClient;
            var disposeClient_ = false;
            try
            {
                using (var request_ = new System.Net.Http.HttpRequestMessage())
                {
                    request_.Method = new System.Net.Http.HttpMethod("GET");
                    request_.Headers.Accept.Add(System.Net.Http.Headers.MediaTypeWithQualityHeaderValue.Parse("text/plain"));
    
                    PrepareRequest(client_, request_, urlBuilder_);
    
                    var url_ = urlBuilder_.ToString();
                    request_.RequestUri = new System.Uri(url_, System.UriKind.RelativeOrAbsolute);
    
                    PrepareRequest(client_, request_, url_);
    
                    var response_ = await client_.SendAsync(request_, System.Net.Http.HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
                    var disposeResponse_ = true;
                    try
                    {
                        var headers_ = System.Linq.Enumerable.ToDictionary(response_.Headers, h_ => h_.Key, h_ => h_.Value);
                        if (response_.Content != null && response_.Content.Headers != null)
                        {
                            foreach (var item_ in response_.Content.Headers)
                                headers_[item_.Key] = item_.Value;
                        }
    
                        ProcessResponse(client_, response_);
    
                        var status_ = (int)response_.StatusCode;
                        if (status_ == 200)
                        {
                            var objectResponse_ = await ReadObjectResponseAsync<string>(response_, headers_, cancellationToken).ConfigureAwait(false);
                            if (objectResponse_.Object == null)
                            {
                                throw new ApiException("Response was null which was not expected.", status_, objectResponse_.Text, headers_, null);
                            }
                            return objectResponse_.Object;
                        }
                        else
                        {
                            var responseData_ = response_.Content == null ? null : await response_.Content.ReadAsStringAsync().ConfigureAwait(false);
                            throw new ApiException("The HTTP status code of the response was not expected (" + status_ + ").", status_, responseData_, headers_, null);
                        }
                    }
                    finally
                    {
                        if (disposeResponse_)
                            response_.Dispose();
                    }
                }
            }
            finally
            {
                if (disposeClient_)
                    client_.Dispose();
            }
        }
```

### .NET Version

6.0.102

**Comment by dougbu** [comment]
@GimpArm thank you for contacting us.

To evaluate this issue, we'd like to understand more about your scenario. Which code generator are you using❔ Is your concern about the generated code, the glue `msbuild` code between the project and the code generator, or the code generator itself❔

Note please do not take my questions as a suggestion this isn't a problem in the dotnet/aspnetcore repo. I'm just unsure what we could change here to address your concerns. We may end up forwarding the issue elsewhere but that's not my initial reaction.

**Comment by GimpArm** [comment]
@dougbu I'm not exactly sure who actually generates the code but now I'm suspecting this is done by NSwag. I thought it was Microsoft.Extensions.ApiDescription.Client but now I'm leaning more towards it being a bug in NSwag.ApiDescription.Client

Just to demonstrate the issue I attached a very simple project. After building the code generated is in obj/swaggerClient.cs and if you look at the generated TestStringAsync() method you see where the accept header is set to "test/plain" (in mine it is line 76) but it still tries to deserialize the response as if it were "application/json" (in mine it is line 101). I attached my generated file too.
[Example.zip](https://github.com/dotnet/aspnetcore/files/8077828/Example.zip)
[swaggerClient.cs.zip](https://github.com/dotnet/aspnetcore/files/8077848/swaggerClient.cs.zip)
